Documentation/Track2TestFramework.md

```diff
@@ -0,0 +1,166 @@
+# Accquring TestFramework
+
+To start using test framework import `sdk\core\Azure.Core\tests\TestFramework.props` into test `.csproj`:
+
+``` xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+...
+  <Import Project="..\..\..\core\Azure.Core\tests\TestFramework.props" />
+...
+
+</Project>
+
+```
+
+# Sync-async tests
+
+The test framework provides an ability to write test using async client methods and automatically run them using sync overloads. To write sync-async client tests inherit from `ClientTestBase` class and use `InstrumentClient` method call to wrap client into a proxy class that would automatically forward async calls to their sync overloads.
+
+``` C#
+public class ConfigurationLiveTests: ClientTestBase
+{
+    public ConfigurationLiveTests(bool isAsync) : base(isAsync)
+    {
+    }
+
+    private ConfigurationClient GetClient()
+    {
+        return InstrumentClient(new ConfigurationClient(...));
+    }
+
+    public async Task DeleteSettingNotFound()
+    {
+        ConfigurationClient service = GetClient();
+
+        var response = await service.DeleteAsync("Setting");
+
+        Assert.AreEqual(204, response.Status);
+        response.Dispose();
+    }
+}
+```
+
+__Limitation__: all method calls/properties that are being used have to be `virtual`.
+
+# Recorded tests
+
+Test framework provides an ability to record HTTP requests and responses and replay them for offline test runs.
+
+To use recorded test functionality inherit from `RecordedTestBase` class and use `Recording.InstrumentClientOptions` method when creating the client instance.
```

Comment:
It'd probably also be good to call out 
- You should only use `Recording.Random` for random values (and you MUST make the same number of random calls in the same order every test run)
- You can use `Recording.Now` and `Recording.UtcNow` if you need certain values to capture the time the test was recorded
- It's possible to add additional recording variables for advanced scenarios (like custom test configuration, etc.)
- You can use `if (Mode == RecordingMode.Playback) { ... }` to change behavior for playback only scenarios (in particular to make polling times instantaneous)